errors.go

```diff
@@ -23,6 +23,11 @@ type ErrorWithDetail interface {
 	DetailMsg() string
 }
 
+type ErrorWithTitle interface {
+	error
+	TitleString() string
```

Comment:
I feel you used this because HTTP Error has a Title field

Why not staying consistent with Detail that had the same issue and used DetailMsg


```suggestion
	TitleMsg() string
```

Reply:
I can yeah. I really don’t like either 😞.

Reply:
Wording is a pain, I know

Reply:
ErrorTitle() string could be a solution maybe

Reply:
I debated on that. I do think DetailMsg should become the same. Albeit breaking, what are your thoughts?

Reply:
I agree in a perfect words ErrorWithTitle and ErrorWithDetail should apply the sane logic.

But world is not prefect. I'm unsure you shlukc fight a breaking change dragon god such minor change